### PR TITLE
[improve](columns)replace logFatal with exception and del some useless code in columns

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
this pr drop all log fatal in columns and replace with throw exception
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 1]
```diff
@@ -127,6 +128,88 @@ class FunctionArraysOverlap : public IFunction {
         return make_nullable(std::make_shared<DataTypeUInt8>());
     }
 
+    Status open(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
```
Comment (eldenmoon) [inline be/src/vec/functions/array/function_arrays_overlap.h:131]: seperate to another pr

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -400,16 +400,31 @@ Status CompactionMixin::execute_compact() {
     data_dir->disks_compaction_score_increment(permits);
     data_dir->disks_compaction_num_increment(1);
 
-    Status st = execute_compact_impl(permits);
+    try {
+        doris::enable_thread_catch_bad_alloc++;
```
Comment (eldenmoon) [inline be/src/olap/compaction.cpp:404]: why not use RETURN_IF_ERROR_OR_CATCH_EXCEPTION？

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run clould_p0

**Comment by amorynan** [comment]
run cloud_p0

File: be/src/common/exception.h [hunk 3]
```diff
@@ -131,3 +131,49 @@ inline const std::string& Exception::to_string() const {
             }                                                                                    \
         }                                                                                        \
     } while (0);
+
+#define HANDLE_EXCEPTION_IF_CATCH_EXCEPTION(stmt, handle_exception)                              \
```
Comment (eldenmoon) [inline be/src/common/exception.h:135]: why add two macros,only one ok

File: be/src/common/exception.h [hunk 4]
```diff
@@ -131,3 +131,49 @@ inline const std::string& Exception::to_string() const {
...
+            if (e.code() == doris::ErrorCode::MEM_ALLOC_FAILED) {                                \
+                return Status::MemoryLimitExceeded(fmt::format(                                  \
+                        "PreCatch error code:{}, {}, __FILE__:{}, __LINE__:{}, __FUNCTION__:{}", \
+                        e.code(), e.to_string(), __FILE__, __LINE__, __PRETTY_FUNCTION__));      \
+            }                                                                                    \
+            return Status::Error<false>(e.code(), e.to_string());                                \
+        }                                                                                        \
+    } while (0);
+
+#define HANDLE_ERROR_IF_CATCH_EXCEPTION_OR_RETURN_ERROR(stmt, handle_error)                      \
```
Comment (eldenmoon) [inline be/src/common/exception.h:158]: using defer

File: be/src/common/exception.h (hunk 3 above)
Comment (eldenmoon) [reply be/src/common/exception.h:135]: rename handle_exception to exception_handler

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p1

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**